src/main/java/com/rapid7/appspider/Base.java

```diff
@@ -268,4 +269,45 @@ private static Object getClassType(HttpResponse response) {
         return null;
     }
 
+    /**
+     * @param apiCall
+     * @param authToken
+     * @param params
+     * @return HttpResponse result from get request to apiCall
+     */
+    private static HttpResponse getResponse(String apiCall, String authToken, Map<String, String> params) {
+        try {
+            //Create HTTP Client
+            HttpClient httpClient = HttpClientBuilder.create().build();
+
+            // Create HttpGet request
+            HttpGet getRequest;
+            if (!params.equals(null)) {
+                URIBuilder uriBuilder = new URIBuilder(apiCall);
+                for (Map.Entry<String, String> entry : params.entrySet()) {
+                    uriBuilder.addParameter(entry.getKey(), entry.getValue());
+                }
+                getRequest = new HttpGet(uriBuilder.build());
+            } else {
+                // Initialized the get request
+                getRequest = new HttpGet(apiCall);
+            }
+
+            // Add the authentication token
+            getRequest.addHeader("Content-Type", "application/x-www-form-urlencoded");
+            getRequest.addHeader("Authorization", "Basic " + authToken);
+
+            // Receive the response from AppSpider
+            HttpResponse getResponse = httpClient.execute(getRequest);
+            return getResponse;
```

Comment:
don't need to create this intermediate object - just return httpClient.execute(getRequest)